### PR TITLE
[YouTube] Improve duration parsing

### PR DESCRIPTION
**Pull request by lrusso96**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

(No API change tho)

I have simplified ad optimized the code to parse the duration of YouTube videos.
In the previous version, the parsing method `parseDurationString` was a bit redundant, making some extra (useless) method calls for short videos.
In particular, there were calls to "sanitize" and check the "0" strings, added within the method.
This code is not only more compact, but only makes the necessary calls to `convertDurationToInt`: e.g., for short videos (<1h), this approach seems twice faster on my device(s).

### CODE REVIEW

**Review by Redirion — APPROVED** [review]
lgtm. I like it. Like a small puzzle that you have solved with the offset.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java [hunk 1]
```diff
@@ -318,38 +318,16 @@ public static int parseDurationString(@Nonnull final String input)
...
-                throw new ParsingException("Error duration string with unknown format: " + input);
-        }
-
-        return ((convertDurationToInt(days) * 24
-                + convertDurationToInt(hours)) * 60
-                + convertDurationToInt(minutes)) * 60
-                + convertDurationToInt(seconds);
+        final int[] units = {24, 60, 60, 1};
+        final int offset = units.length - splitInput.length;
+        if (offset < 0) {
```
Comment (acrodemocide) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:323]: In the case statement being replaced, the valid values were 1, 2, 3, and 4. Consider changing this if statement to:

if (offset < 1 || offset > 4)

This should preserve the same logic that existed in the if statement.

**Review by acrodemocide — COMMENTED** [review]
Consider changing the if statement commented on to preserve the logic being replaced for the case statement.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java (hunk 1 above)
Comment (lrusso96) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:323]: > In the case statement being replaced, the valid values were 1, 2, 3, and 4. Consider changing this if statement to:
> 
> if (offset < 1 || offset > 4)
> 
> This should preserve the same logic that existed in the if statement.

@acrodemocide `splitInput.length` is a number strictly greater than 0, while `units.length` is equal to (the magic number) 4.
Then `offset` cannot be a number strictly greater than 4.

Comment (acrodemocide) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:323]: Thank you for clarifying.

**Review by acrodemocide — APPROVED** [review]
Comment resolved

**Pull request merged**